### PR TITLE
Bind SQL invokable object to the provider

### LINKED ISSUE #29: SQL specified by Sql annotation/attribute
* インジェクションされる変数名と同じプロパティ名の`@Sql`がSQL文の特定に使われます。
* callableの代わりに`InvokeInterface`インターフェイスを使います。

PHP 7.x
```php
class TodoRepository
{
    /**
     * @Sql("todo_insert")
     */
    public $todoCreate;

    /**
     * @Sql("todo_item_by_id")
     */
    public $todoItem;

    /**
     * @Sql("todo_list")
     */
    public $todoList;

    public function __construct(
        InvokeInterface $todoCreate,
        RowInterface $todoItem,
        RowListInterface $todoList
    ){
        $this->todoCreate = $todoCreate;
        $this->todoItem = $todoItem;
        $this->todoList = $todoList;
    }
}
```

`$todoCreate`には`todo_insert.sql`SQL実行オブジェクトがインジェクトされます。

PHP8.x
```php
<?php

declare(strict_types=1);

namespace Ray\Query;

use Ray\Query\Annotation\Sql;

class TodoRepository
{
    public function __construct(
        #[Sql('todo_insert')] public readonly InvokeInterface $todoCreate,
        #[Sql('todo_item_by_id')]  public readonly RowInterface $todoItem,
        #[Sql('todo_list')]  public readonly RowListInterface $todoList
    ){}
}
```
現状の`@Named`の煩雑さと開発時の実行速度を改善します。

### PR DESCRIPTION
**Pull request by koriym**
Closes #29 

## Summary by CodeRabbit

- **New Features**
  - Introduced new classes and interfaces for handling SQL queries and exceptions.
  - Added new dependencies for enhanced functionality.

- **Improvements**
  - Updated PHP version and dependencies for better compatibility.
  - Enhanced SQL query handling with new public properties and methods.
  - Improved dependency injection configuration for SQL-related classes.

- **Tests**
  - Added new test cases for SQL query modules and providers.
  - Improved assertions and array checks in existing test methods.

- **Chores**
  - Updated CI and static analysis workflow configurations.

### CODE REVIEW

**Comment by koriym** [comment]
@coderabbitai reviw

File: src/SqlFinder.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        SqlDir $sqlDir
+    ) {
+        $this->reader = $reader;
+        $this->sqlDir = $sqlDir;
+    }
+
+    public function __invoke(ReflectionParameter $param): string
+    {
+        /** @var ?Sql $sqlAnnotation */
+        $sqlAnnotation = $this->reader->getParametrAnnotation($param, Sql::class);
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/SqlFinder.php:37]: Fixed.

**Comment by koriym** [comment]
@coderabbitai review

**Comment by koriym** [comment]
https://github.com/coderabbitai review

**Comment by koriym** [comment]
@coderabbitai review

**Review by jingu — APPROVED** [review]
確認しました。問題ないと思います。
確かにこれなら大量のSQLファイルがあったとしても、開発時のファイル読み込みコストを削減できそうで👍です。

**Pull request merged**